packages/components/pie-modal/src/index.ts

```diff
@@ -44,20 +44,57 @@ export class PieModal extends RtlMixin(LitElement) {
                         variant="ghost-tertiary"
                         class="c-modal-closeBtn"></pie-icon-button>
                 </header>
-                <article>
-                    <div class="c-modal-content">
-                        <slot></slot>
-                    </div>
+                <article class="c-modal-content">
+                    <slot></slot>
                 </article>
             </dialog>
         `;
     }
 
+    updated () {
+        this._handleDialogLightDismiss();
+    }
+
+    /**
+     * We require this because toggling the prop `isOpen` itself won't
+     * allow the dialog to open in the correct way (with the default background),
+     * the method `showModal()` needs to be invoked.
+     *
+     * https://lit.dev/docs/components/lifecycle/#willupdate
+     *
+     * @param changedProperties
+     */
+    willUpdate (changedProperties: Map<string, any>) {
```

Comment:
my understand is `willUpdate` is to prep data ahead of an update. `updated` is for performing actions during/after. is `willUpdate` definitely right here?

Reply:
The event listener that gets attached needs to be prepped before the update otherwise there is an error. You can't really use the connected or disconnected methods on modal element as you will get an error there too. This was something i was going to ask the lit team about until i read about willUpdate.

Reply:
![Screenshot 2023-06-22 at 16 36 53](https://github.com/justeattakeaway/pie/assets/2299779/14bf8c52-378a-40f8-be60-0b1c25160101)

Using `update`

Reply:
Because you are attaching it to a document not the dom shadow element so you wont get that error.